Fix#4864 - On mobile, two icons are visible on the exploration/collection cards at the same time.

srikar0896 opened the pull request:
Fixes #4864 - On mobile, two icons are visible on the exploration/collection cards at the same time.

Expects to work in this way

- [x]  Check icon if the exploration/collection has been completed.
- [x] Spinner icon if the user is partway through the exploration/collection.
- [x] Ellipses icon if the user has not visited the exploration/collection before.
- [x] Clock icon if the exploration/collection is present in the playlist.

nithusha21 reviewed (approved):
These changes LGTM. Just one thing, I cannot add to playlist after searching for an exploration using the search bar. Probably not a blocking bug, but something we should take a look at in the future. What do you say?

srikar0896 commented:
Yes, I was also thinking of fixing that in this commit itself. I think it's important to fix this issue.I see the problem is the container-is-narrow attribute is being passed in exploration-summary-tile/collection-summary-tile so that it can be used in the learner-dashboard-icons directive. 
![image](https://user-images.githubusercontent.com/17567875/38755449-5cf17dc8-3f83-11e8-97a8-34d1cbe1cbde.png)

But in activity-tiles-grid directive used in the search-results directive, the container-is-narrow attribute is not being passed.So passing that attribute would fix this.
Shall I fix that in this PR?

seanlip commented:
I think fixing this in this commit sounds like a good idea!

srikar0896 committed:
Changed the condition for visibility of the ellipsis icon.

srikar0896 committed:
Fix - Cannot add an activity to playlist from the search results.

seanlip commented:
Hi @srikar0896 -- did you actually intend to close this PR?

seanlip commented:
Also I'm a bit confused, all the files you changed disappeared?

srikar0896 commented:
@seanlip No, Actually I have added a commit with a spelling mistake in the commit message, So to correct that I have deleted the last commit by 
git reset --soft HEAD^
git push origin +HEAD^:icon-visibility-fix.
That command must delete the latest commit but I was confused how the initial files were also deleted.

srikar0896 commented:
@seanlip Sorry, Actually I have not added the changes in the directive controller to the latest commit, So to add that file to the same commit I have deleted the last commit by
git reset --soft HEAD^
git push origin +HEAD^:icon-visibility-fix.
That command must delete the latest commit but I was confused how the initial files were also deleted.
I have re-added those commits.Thanks

seanlip commented:
OK, thanks. @nithusha21 could you please take another look when you get a chance?

nithusha21 reviewed (approved):
Thanks @srikar0896. The changes look good to me 👍

DubeySandeep commented on core/templates/dev/head/pages/library/ActivityTilesInfinityGridDirective.js:
```diff
@@ -24,8 +24,8 @@ oppia.directive('activityTilesInfinityGrid', [
-        '$scope', '$rootScope', 'SearchService',
-        function($scope, $rootScope, SearchService) {
+        '$scope', '$rootScope', 'SearchService', 'WindowDimensionsService',
+        function ($scope, $rootScope, SearchService, WindowDimensionsService) {
```
Do we need space here? [before `(`]

DubeySandeep commented on core/templates/dev/head/pages/library/ActivityTilesInfinityGridDirective.js:
```diff
@@ -52,6 +52,16 @@ oppia.directive('activityTilesInfinityGrid', [
+          $scope.libraryWindowIsNarrow = (
+            WindowDimensionsService.getWidth() <= libraryWindowCutoffPx);
+
+          WindowDimensionsService.registerOnResizeHook(function () {
```
Ditto. [Remove space before `(`]

srikar0896 replied:
Done.Thanks

srikar0896 committed:
Removed space.
